web3.js/src/connection.ts

```diff
@@ -297,6 +302,29 @@ export type BlockheightBasedTransactionConfirmationStrategy = {
   signature: TransactionSignature;
 } & BlockhashWithExpiryBlockHeight;
 
+/**
+ * @internal
+ */
+function createRPCError(
+  error: Readonly<{code: unknown; message: string; data?: any}>,
+  customMessage: string,
+): SolanaJSONRPCError {
+  const code = coerceSolanaJSONRPCErrorCode(error.code);
+  return new SolanaJSONRPCError(`${customMessage}: ${error.message}`, code);
+}
+
+/**
+ * @internal
+ */
+function coerceSolanaJSONRPCErrorCode(
```

Comment:
I don't think we should coerce the error code because the rpc server might add new error codes and it'd be useful for clients to debug which unexpected error code they received